src/game_state.c

```diff
@@ -59,7 +59,7 @@ u32 sub_08152A18(void) {
     return 1;
 }
 
-struct GameState* sub_08152B00(GameStateFunc arg0, u16 arg1, u16 arg2, u16 arg3, GameStateFunc2 arg4) {
+struct GameState* CreateState(GameStateMain arg0, u16 arg1, u16 arg2, u16 arg3, GameStateDestructor arg4) {
```

Comment:
`GameStateCreate`
The reason why I'm changing the order is because you named `GameStateInit` instead of `InitGameState`